Fix bug in proxy code

msgodf opened the pull request:
When `:download_proxy_host` is set in the config, lein-embongo fails, with a stack trace
```
SEVERE: Build ArtifactStore(useCache:true)
Download GenericVersion{2.6.4}:Linux:B64 START
java.lang.IllegalArgumentException: No matching method found: InetSocketAddress for class java.lang.String
 at clojure.lang.Reflector.invokeMatchingMethod (Reflector.java:53)
    clojure.lang.Reflector.invokeInstanceMethod (Reflector.java:28)
    leiningen.embongo$add_proxy_selector_BANG_$fn__8039.invoke (embongo.clj:19)
    leiningen.embongo.proxy$java.net.ProxySelector$ff19274a.select (:-1)
    sun.net.www.protocol.http.HttpURLConnection.plainConnect0 (HttpURLConnection.java:1097)
    sun.net.www.protocol.http.HttpURLConnection.plainConnect (HttpURLConnection.java:997)
    sun.net.www.protocol.http.HttpURLConnection.connect (HttpURLConnection.java:931)
    sun.net.www.protocol.http.HttpURLConnection.getInputStream0 (HttpURLConnection.java:1511)
    sun.net.www.protocol.http.HttpURLConnection.getInputStream (HttpURLConnection.java:1439)
    de.flapdoodle.embed.process.store.Downloader.download (Downloader.java:74)
    de.flapdoodle.embed.process.store.ArtifactStore.checkDistribution (ArtifactStore.java:53)
    de.flapdoodle.embed.process.store.CachingArtifactStore.checkDistribution (CachingArtifactStore.java:54)
    de.flapdoodle.embed.process.runtime.Starter.prepare (Starter.java:55)
    leiningen.embongo$start_mongo.invoke (embongo.clj:47)
    leiningen.embongo$embongo.doInvoke (embongo.clj:62)
````
The bug is caused by trying to call InetSocketAddress as a method on String, rather than instantiating it.

joelittlejohn commented:
:astonished:

Thanks Mark!!

joelittlejohn commented:
I haven't published a new release of this since clojars turned off scp. I guess I can't hold off any longer xD

joelittlejohn commented:
I actually think in recent versions of lein there's no need to use a proxy if you have environment vars like `http_proxy` set. You might already know this, but just in case.

msgodf commented:
Sorry that it's such a minor change :) I bumped the embedded MongoDB version that we were using and found proxy issues, fixed them for myself with environment vars - then eventually found (and fixed) the bug when someone else had issues with the proxy. I guess the proxy code could be removed, but there might be a use case where people want a proxy for the MongoDB download but nothing else.

msgodf commented:
I also have a dev version of lein-embongo where I've updated the config code to work with the newest flapdoodle embedded Mongo, but it doesn't add any functionality to lein-embongo itself yet. I've been hoping that the extracted binary cache feature mentioned in flapdoodle-oss/de.flapdoodle.embed.mongo/issues/90 gets implemented.

msgodf commented:
(I mainly mention the above in case you'd like me to PR that change before you push a new release)

joelittlejohn commented:
Yeah, that would be great go for it. Cheers
On 16 Feb 2015 20:55, "Mark Godfrey" <notifications@github.com> wrote:

> (I mainly mention the above in case you'd like me to PR that change before
> you push a new release)
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/joelittlejohn/lein-embongo/pull/6#issuecomment-74571144>
> .
>

joelittlejohn commented:
@msgodf I've released 0.2.2 after moving over to the new release/sign/deploy steps for Clojars. I also did a bit of tidying up and ended up upgrading the flapdoodle API too so don't worry about the PR. Very easy to release again now so if you want to implement the extracted binary cache feature in the flapdoodle API then just shout and I'll bump the version again.
